lib: peer_manager: port lesc module to use PSA APIs

anhmolt opened the pull request:
Update the LE secure part of the peer_manager to use PSA APIs. In SDK5 they were using the nrf_crypto libraries.

anhmolt committed:
lib: peer_manager: port lesc module to use PSA APIs

Update the LE secure part of the peer_manager to use PSA APIs.
In SDK5 they were using the nrf_crypto libraries.

Signed-off-by: Andreas Moltumyr <andreas.moltumyr@nordicsemi.no>

lemrey commented on lib/peer_manager/modules/nrf_ble_lesc.c:
```diff
@@ -56,59 +53,68 @@ static bool m_lesc_oobd_own_generated;
+}
+
+/* Invert byte array. */
+static void byte_order_invert(uint8_t *data, size_t len)
```
You can include `zephyr/include/zephyr/sys/byteorder.h` and re-use `sys_mem_swap()`

anhmolt replied:
Thanks! Updated.

lemrey commented on lib/peer_manager/modules/nrf_ble_lesc.c:
```diff
@@ -56,59 +53,68 @@ static bool m_lesc_oobd_own_generated;
+/* Convert an ECC (secp256r1) public key from between big-endian and little-endian.
+ * The two coordinates are converted individually.
+ */
+static void ecc_public_key_byte_order_invert(const uint8_t *raw_key_in, uint8_t *raw_key_out)
```
and here you could perhaps use

```
sys_memcpy_swap(x_out, x_in, COORDINATE_SIZE);
sys_memcpy_swap(y_out, y_in, COORDINATE_SIZE);
```

from the `byteorder.h` header too

anhmolt replied:
Thanks! Seems to work well. Updated.

Vge0rge commented on lib/peer_manager/modules/nrf_ble_lesc.c:
```diff
@@ -118,35 +106,55 @@ uint32_t nrf_ble_lesc_keypair_generate(void)
+		return NRF_ERROR_INTERNAL;
+
+	} else if (pub_key_len != sizeof(pub_key)) {
+		LOG_ERR("psa_export_public_key() unexpected output length %d", pub_key_len);
```
I don't think that you need a run time check for this to be honest.  The PSA apis specify both the size of the  output and the representation. So this seems redundant and the check below about the pub_key first byte being 0x4 as well.

anhmolt replied:
Thanks. Removed the extra checks.

Vge0rge commented on lib/peer_manager/modules/nrf_ble_lesc.c:
```diff
@@ -56,59 +54,49 @@ static bool m_lesc_oobd_own_generated;
-	uint32_t err_code;
-	size_t public_len = NRF_CRYPTO_ECC_SECP256R1_RAW_PUBLIC_KEY_SIZE;
+	psa_status_t status;
+	uint8_t pub_key[sizeof(uint8_t) + BLE_GAP_LESC_P256_PK_LEN];
```
Nit but if you want you can  use this define here for the size [PSA_EXPORT_PUBLIC_KEY_OUTPUT_SIZE](https://arm-software.github.io/psa-api/crypto/1.3/api/keys/management.html#c.PSA_EXPORT_PUBLIC_KEY_OUTPUT_SIZE)(type, bits)

anhmolt replied:
Thanks. Updated.

anhmolt committed:
ble_hrs: add crypto configuration for LE secure connection support

Add the required crypto configuration for using the LE Secure
Connection module of the peer manager. This is done to
show what is required.

Signed-off-by: Andreas Moltumyr <andreas.moltumyr@nordicsemi.no>

anhmolt commented:
Did some changes to support static key slots, not just mbedtls heap. Saves about 1 KB of ROM and 1.8 KB of RAM.
Also added the required configurations for LE secure connection to ble_hrs sample for testing.

@MirkoCovizzi It is ready to be merged on my side.
